src/condor_utils/proc_family_direct_cgroup_v1.cpp

```diff
@@ -346,30 +332,62 @@ ProcFamilyDirectCgroupV1::get_usage(pid_t pid, ProcFamilyUsage& usage, bool /*fu
 	}
 
 	// These are in USER_HZ, which is always (?) 100 hz
-	uint64_t user_hz = 0;
-	uint64_t sys_hz  = 0;
+	user_cpu_hz = 0;
+	sys_cpu_hz  = 0;
 
 	char word[128]; // max size of a word in cpu.stat
 	while (fscanf(f, "%s", word) != EOF) {
 		// if word is usage_usec
 		if (strcmp(word, "user") == 0) {
 			// next word is the user time in microseconds
-			if (fscanf(f, "%ld", &user_hz) != 1) {
+			if (fscanf(f, "%ld", &user_cpu_hz) != 1) {
 				dprintf(D_ALWAYS, "Error reading user_usec field out of cpu.stat\n");
 				fclose(f);
 				return false;
 			}
 		}
 		if (strcmp(word, "system") == 0) {
 			// next word is the system time in microseconds
-			if (fscanf(f, "%ld", &sys_hz) != 1) {
+			if (fscanf(f, "%ld", &sys_cpu_hz) != 1) {
 				dprintf(D_ALWAYS, "Error reading system_usec field out of cpu.stat\n");
 				fclose(f);
 				return false;
 			}
 		}
 	}
 	fclose(f);
+	return true;
+}
+
+bool
+ProcFamilyDirectCgroupV1::get_usage(pid_t pid, ProcFamilyUsage& usage, bool /*full*/)
+{
+	// DaemonCore uses "get_usage(getpid())" to test the procd, ignoring the usage
+	// even if we haven't registered that pid as a subfamily.  Or even if there
+	// is a procd.
+
+	if (pid == getpid()) {
+		return true;
+	}
+
+	std::string cgroup_name = cgroup_map[pid];
+
+	// Initialize the ones we don't set to -1 to mean "don't know".
+	usage.block_reads = usage.block_writes = usage.block_read_bytes = usage.block_write_bytes = usage.m_instructions = -1;
+	usage.io_wait = -1.0;
+	usage.total_proportional_set_size_available = false;
+	usage.total_proportional_set_size = 0;
+
+	stdfs::path cgroup_root_dir = cgroup_mount_point();
+	stdfs::path leaf            = cgroup_root_dir / "cpu,cpuacct" / cgroup_name;
+
+	uint64_t user_hz = 0;
+	uint64_t sys_hz = 0;
+
+	get_user_sys_cpu_hz(cgroup_name, user_hz, sys_hz);
+
+	user_hz -= start_user_cpu_hz;
+	sys_hz  -= start_sys_cpu_hz;
```

Comment:
If get_user_sys_cpu_hz() fails, then we can end up with negative cpu values. This may affect the v2 version as well.